components/Nav/Nav.test.tsx

```diff
@@ -0,0 +1,64 @@
+import { render, screen } from '@testing-library/react';
+import Nav from './Nav';
+
+const mockPush = jest.fn();
+const mockUsePathname = jest.fn();
+
+const mockUseAuthContext = {
+  logoutAccount: jest.fn(),
+};
+
+jest.mock('next/navigation', () => ({
+  useRouter() {
+    return {
+      push: mockPush,
+    };
+  },
+  usePathname() {
+    return mockUsePathname();
+  },
+}));
+
+jest.mock('../../context/AuthContextProvider', () => ({
+  useAuthContext() {
+    return {
+      ...mockUseAuthContext,
+    };
+  },
+}));
+
+describe('Nav', () => {
+  beforeAll(() => {
+    jest.clearAllMocks();
+  });
+  it('it should be hidden when path is /register', () => {
```

Comment:
m: We need to have a default test that just checks the default state of the component. Ensure the title is there, LogoNav, and logout button. 

Need to also add a test to ensure that when the user clicks on the logout button, a mocked function logs the user out and redirects them to the login page.

Reply:
Got it, so an additional test to check the default state of the component (ensuring title, LogoNav, and logout button are present) and another test for the logout button (user clicks logout button, mock function for logout, the user should be back at the /login page). I will add. Thank you again, Shashi!